python/mlcroissant/mlcroissant/_src/datasets.py

```diff
@@ -114,15 +117,16 @@ def __iter__(self):
         Warning: at the moment, this method yields examples from the first explored
         record_set.
         """
-        operations = self.dataset.operations.operations
+        operations = self._filter_interesting_operations()
         if self.debug:
             graphs_utils.pretty_print_graph(operations)
         # Downloads can be parallelized, so we execute them in priority.
         execute_downloads(operations)
         # We can stream the dataset iff the operation graph is a path graph (meaning
         # that all operations lie on a single straight line, i.e. have an
         # in-degree of 0 or 1. That means that the operation graph is a single line
-        # (without external joins for example).
+        # (without external joins for example). We only consider the operations that
+        # are useful to produce the `ReadFields(record_set)`.
```

Comment:
> We only consider the operations that are useful to produce the `ReadFields(record_set)`

This depends on _filter_interesting_operations() no?
If yes, then maybe this comment could be put above?